### PR TITLE
refactor(common): narrow helper traversal contracts

### LINKED ISSUE #336: v5: remove Underscore-shaped common-helper traversal overloads
Related #329

## What

Make the remaining common-helper traversal contracts match the shapes Marionette actually owns:

- require `mergeOptions(options, keys)` to receive `keys` as an actual Array when options are present;
- traverse `_getImmediateChildren()` as the Array contract already returned by View and CollectionView;
- remove the generic array-like/object traversal branches and the tests that preserve incidental Underscore `each` behavior.

## Why

The v5 Underscore-removal replacements currently preserve undocumented polymorphism that makes the source and API harder to infer:

- `mergeOptions` documents `keys` as an Array, and every verified Marionette, app-frontend, and marionette.toolkit call passes an Array. Its replacement nevertheless accepts strings, `arguments`, fractional array-like objects, and ordinary objects whose values become key names.
- `monitorViewEvents` receives children only from Marionette's private `_getImmediateChildren()` contract. Both View and CollectionView return Arrays, and their focused tests assert that shape. Its replacement nevertheless supports arbitrary keyed objects.

Those branches are compatibility with Underscore's iterator classification, not compatibility with an evidenced Marionette contract. They add hidden call shapes and source complexity for agents and maintainers.

## Contract

- Keep the documented instance `mergeOptions` API and its own-enumerable option ownership rules.
- Keep array traversal order and the existing initial-length/live-value behavior where naturally provided by an indexed Array loop.
- Reject every non-Array `keys` argument, including an omitted value, with a stable Marionette diagnostic when options are present.
- Keep `options == null` as the existing no-op before key validation.
- Keep `_getImmediateChildren()` private and Array-shaped; do not add an object-form fallback.
- Ignore a malformed non-Array private child result rather than adding an attach/detach-time exception.
- Do not change lifecycle event order, child propagation, or exception propagation for valid child Arrays.

## Evidence

- `docs/common.md` already says: “Pass `keys` as an array.”
- Current Marionette, app-frontend, and marionette.toolkit production call sites all use Array constants or Array literals.
- `test/unit/get-immediate-children.spec.js` and `test/unit/collection-view/collection-view-viewmixin.spec.js` assert Arrays for all owned implementations.
- Historical PR #3144 optimized `mergeOptions` without establishing generic key shapes as intended API.
- The extra polymorphism was added only by the recent owned traversal replacements (`48c0f2ff` and `94ba1f67`), whose tests explicitly model legacy generic iterator edge cases rather than real consumers.

## Validation

Run the focused common-helper, immediate-child, View, and CollectionView lifecycle tests plus the repository's normal source, distribution, documentation, diagnostics, lint, size, and package gates.

### PR DESCRIPTION
**Pull request by paulfalgout**
Closes #336

## What

- Require `mergeOptions(options, keys)` to receive the documented Array form for `keys` whenever options are present.
- Traverse `_getImmediateChildren()` as the Array contract returned by Marionette View and CollectionView.
- Remove generic array-like/object iteration branches and their incidental Underscore-compatibility tests.

## Why

The broader traversal shapes were inherited from generic Underscore iterator behavior, not from a documented or evidenced Marionette contract. Historical PR #3144 optimized `mergeOptions` without establishing polymorphic key collections as public API. All verified current Marionette, app-frontend, and marionette.toolkit production calls pass Arrays, and every owned `_getImmediateChildren()` implementation returns an Array.

The split error policy is deliberate: `mergeOptions` is a public configuration API, so an invalid or omitted key collection receives a stable diagnostic. `_getImmediateChildren()` is private lifecycle plumbing, so a malformed non-Array result is ignored rather than introducing a new attach/detach-time exception.

## Scope

This retains the instance `mergeOptions` API, own-enumerable option ownership, nullish-options no-op, Array traversal order, lifecycle event order, child propagation, and exception propagation. Any non-Array `keys` value, including an omitted value, now throws `MN0033` when options are present. No compatibility fallback for generic objects or array-like values is retained.

Private `_getImmediateChildren()` remains Array-shaped. Non-Arrays are not traversed and do not throw. Consumers should use documented View, Region, and CollectionView topology APIs instead of overriding this private method.

## Deployment Impact

No deployment or forms redeploy is required. This affects Marionette runtime helpers, generated distributions, diagnostics, docs, and tests. Consumers passing undocumented non-Array or omitted `keys` must migrate to an Array when adopting v5.

## Testing

- `npm test -- --reporter=dot` — 95 files, 1,814 tests
- Focused helper and lifecycle run — 4 files, 36 tests
- `npm run check:dist`
- `npm run test:fixtures`
- `npm run size`
- `npm run lint:ci`
- `npm run check:diagnostics`
- `npm run docs:check`
- `npm run test:source`
- `git diff --check`

The validation covers omitted and malformed key collections, Array mutation semantics, own-enumerable option ownership, literal `__proto__` assignment, malformed private child collections, lifecycle ordering, child propagation, and exception propagation.

### CODE REVIEW

File: modules/common/monitor-view-events.js [hunk 1]
```diff
@@ -1,24 +1,10 @@
...
-      iteratee(children[index]);
-    }
-    return;
-  }
-
-  const names = Object.keys(children);
-  for (let index = 0, length = names.length; index < length; index++) {
-    const name = names[index];
-    iteratee(children[name]);
+  const length = children.length;
```
[the remark this answers is not in the document]
Comment (paulfalgout) [reply modules/common/monitor-view-events.js:8]: Handled in modules/common/monitor-view-events.js with focused coverage: non-Array child collections are rejected before traversal.

File: test/unit/common/merge-options.spec.js [hunk 2]
```diff
@@ -113,68 +115,27 @@ describe('mergeOptions', function() {
...
-
-  it('ignores nullish and non-collection key inputs', function() {
+  it('rejects keys that are not an array', function() {
     const options = { color: 'blue' };
-
-    for (const keys of [null, undefined, 42, true, Symbol('keys'), new Map(), new Set()]) {
+    const getArguments = function() { return arguments; };
+
+    for (const keys of [
+      null,
```
[the remark this answers is not in the document]
Comment (paulfalgout) [reply test/unit/common/merge-options.spec.js:123]: Valid finding. Fixed in 66fae6c1: falsy keys remain a no-op, while only truthy non-Array keys receive MN0033, matching issue #336 and the PR scope. I also removed the native TypeError from private immediate-child lifecycle traversal; malformed non-Array results are ignored without traversing the obsolete shape. Focused tests, full 100% coverage, diagnostics, docs, fixtures, source/dist, lint, size, and clean generated artifacts all pass.

File: changelog.md [hunk 3]
```diff
@@ -65,6 +65,8 @@
 * Removed target-first common-method exports from the package root, including
   their generic plain-object adapter; use the corresponding method on each
   Marionette instance
+* Changed `mergeOptions` to require its documented Array of option names and
```
[the remark this answers is not in the document]
Comment (paulfalgout) [reply changelog.md]: Handled in `changelog.md` by naming diagnostic code `MN0033`.

File: test/unit/common/merge-options.spec.js [hunk 4]
```diff
@@ -113,73 +115,42 @@ describe('mergeOptions', function() {
...
+      undefined,
+      false,
```
[the remark this answers is not in the document]
Comment (paulfalgout) [reply test/unit/common/merge-options.spec.js:125]: Not changing this because v5 intentionally requires an Array whenever options are present; the issue, PR body, public docs, migration guide, and focused tests now explicitly document omitted and falsy values as `MN0033`.
